Firebase/InstanceID/FIRInstanceIDCheckinService.m

```diff
@@ -69,12 +70,16 @@ - (void)dealloc {
 }
 
 - (void)checkinWithExistingCheckin:(FIRInstanceIDCheckinPreferences *)existingCheckin
+                 firebaseUserAgent:(NSString *)firebaseUserAgent
                         completion:(FIRInstanceIDDeviceCheckinCompletion)completion {
   _FIRInstanceIDDevAssert(completion != nil, @"completion required");
 
   NSURL *url = [NSURL URLWithString:kDeviceCheckinURL];
   NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url];
+
   [request setValue:@"application/json" forHTTPHeaderField:@"content-type"];
+  [request setValue:firebaseUserAgent forHTTPHeaderField:kFIRInstanceIDFirebaseUserAgentKey];
```

Comment:
Does firebaseUserAgent change a lot or need to query at the beginning of IID initialization? Can you just call [[firApp class] firebaseUserAgent] here instead of passing it in through all the chains??

Reply:
@chliangGoogle That was my first thought - just call [FIRApp firebaseUserAgent] from `FIRInstanceIDCheckinService`. Though, in this case we will introduce an additional dependency for `FIRInstanceIDCheckinService` as well as an additional responsibility - to retrieve user agent. 
It looks to me that none of the classes from the chain should depend on `FIRApp` or be responsible for retrieving the user-agent string.

Please let me know if it doesn't look right.

Reply:
I don't think that's bigger drawback, your code could be much simpler tho by adding a dependency of FIRApp, especially InstanceID depends on core anyway.